File: color2gray.m
%gray = .21 * red + .72 * green + .07 * blue;
function [new_image original_image] = color2gray(image)
  original_image = image;
  [row, col, rgb] = size(image);
  new_image = uint8(zeros(row, col, rgb));
  
  % To display an image in gray scale, 
  %   set the rgb to the same values.
  
  for i = 1:row
      for j = 1:col
        new_image(i,j,1) = image(i,j,1)*.21 + ...
            image(i,j,2)*.72 + ...
            image(i,j,3)*.07;
        new_image(i,j,2) = new_image(i,j,1);
        new_image(i,j,3) = new_image(i,j,1); 
      end
  end
  new_image = uint8(new_image);
  
end

File: color2gray.c
// gray = .21 * red + .72 * green + .07 * blue;

#include <stdio.h>
#include <string.h>

int main(void)
{
  volatile int * iRead_Data1 = (int *) 0x00000000; // input from read_data1
  volatile int * iRead_Data2 = (int *) 0x00000000; // input from read_data2
  volatile int * oRead_Data1 = (int *) 0x00000000; // output to read_data1
  volatile int * oRead_Data2 = (int *) 0x00000000; // output to read_data2
  
  int red, green, blue;
  
  // Separate inputs into rgb
  
  // Do the multiplication
  // Store into the output
  return 0;
}
